Cinema Journal/Controller/ViewController.swift

```diff
@@ -51,24 +61,41 @@ extension ViewController : UITableViewDataSource {
         return results.count
     }
     
+    
+    
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: MovieCellTableViewCell.cellIdentifier, for: indexPath) as! MovieCellTableViewCell
         guard indexPath.row < results.count  else { fatalError("cell index out of bounds at movies tableview") }
-        
         cell.configureCell(results[indexPath.row])
+        
         return cell
     }
     
-    //MARK: - UIAlert
-    func showAlert() {
-        let ac = UIAlertController(title: NSLocalizedString("network_error_title", comment: ""), message: NSLocalizedString("fetch_failure", comment: ""), preferredStyle: .alert)
-        let okButton = UIAlertAction(title: "Ok", style: .cancel, handler: nil)
-        ac.addAction(okButton)
-        present(ac, animated: true, completion: nil)
+}
+
+extension ViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        performSegue(withIdentifier: segueID, sender: indexPath)
+        tableView.deselectRow(at: indexPath, animated: true)
     }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+         if segue.identifier == segueID {
+             guard let destinationVC = segue.destination as? DetailMovieVC else {return}
+            let row = (sender as! NSIndexPath).row
+            let model = results[row]
+            DispatchQueue.main.async {
```

Comment:
Segue perfomed always on main queue, dispatch is redundant

Reply:
Without _.main.async_ **configureModel(_)** would execute before network call, causing unwrapping optional value, as a result the app would crash at runtime

Reply:
You're not making network call, you preserve array of already objects. If there is segue perfomed — you have an object in array already guaranteed